### PR TITLE
78 create slave master i2c

### PR DESCRIPTION
**Pull request by kierenblack**
Created basic code to have an arduino as a slave receiving data from sensors and a TTGO as the master that sends to LoRa.

### CODE REVIEW

File: sensors/i2c_master_slave/mega_slave/mega_slave.ino [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    // Wind speed calculation
+    windSpeed = windPulseCount * 2.4 / 10.0;  // Convert pulses to m/s (assuming 2.4 m/s for 1 pulse per second)
+    windPulseCount = 0;  // Reset wind pulse count after each update
+    previousWindUpdateMillis = currentMillis;  // Update the last wind update time
+
+    // Rainfall calculation (each pulse represents 0.2794 mm of rainfall)
+    rainfall = rainPulseCount * 0.2794;
+
+    // Print sensor values to the serial monitor
```
Comment (SamanthaCra) [inline sensors/i2c_master_slave/mega_slave/mega_slave.ino:72]: If we don't need these they can probably be removed

File: sensors/i2c_master_slave/mega_slave/mega_slave.ino [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+// ISR for wind sensor on D2/INT4
+void windISR() {
+  windPulseCount++;  // Increment wind pulse count
+}
+
+// ISR for rain sensor on D3/INT5
+void rainISR() {
+  rainPulseCount++;  // Increment the rainfall pulse count
+}
```
Comment (SamanthaCra) [inline sensors/i2c_master_slave/mega_slave/mega_slave.ino:145]: Does this send all of the data as one payload or multiple?

**Review by SamanthaCra — COMMENTED** [review]
Just wondering if this seperates the payload or not and if it adds the f port?

File: sensors/i2c_master_slave/mega_slave/mega_slave.ino (hunk 1 above)
Comment (kierenblack) [reply sensors/i2c_master_slave/mega_slave/mega_slave.ino:72]: this is needed to measure rainfall

File: sensors/i2c_master_slave/mega_slave/mega_slave.ino (hunk 2 above)
Comment (kierenblack) [reply sensors/i2c_master_slave/mega_slave/mega_slave.ino:145]: It resets every 24 hours to do daily rainfall

**Pull request merged**